### DISCUSSION TITLE
Limiting the amount of $ that can be used per APIKey

### DISCUSSION THREAD

**Discussion by AntDX316** [deepgram/community#705] [General help] [answered]
Can you make it where an apikey is limited to a set amount of $ usage per
month just in case the API key gets compromised or is used for too long resulting in the wipe out of the entire credits remaining creating distrust with deepgram?

I want to be able to say allow only $5 per key for rotation please.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment, marked as answer]
@AntDX316  👋  Thanks for the suggestion. We currently don't offer a feature to rate limit an API Key by a set dollar value. But that has been requested before by other users of Deepgram. 

 I'll share this with our Product Team for feedback.

**Comment by SandraRodgers** [comment]
You can make API keys that expire after a certain time limit. That could be one way to have some control over their usage. Read more in this blog post: https://deepgram.com/learn/protecting-api-key
